From 35c3c3eeba217b94b2bdd210ba3b51b24920986c Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Thu, 7 Dec 2017 22:42:34 +0100
Subject: [PATCH] Settings: Hide Notification Dots on low RAM devices

 * As the feature is not available for low RAM devices,
    hide it from the Settings to avoid confusion

 * Acquire ActivityManager service to read isLowRamDevice

Change-Id: I34a6ee1b4838f31194c89327e301d4fd39670c02
Signed-off-by: Adrian DC <radian.dc@gmail.com>
---
 src/com/android/launcher3/SettingsActivity.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/SettingsActivity.java b/src/com/android/launcher3/SettingsActivity.java
index 32c198ab0..6ea518f74 100644
--- a/src/com/android/launcher3/SettingsActivity.java
+++ b/src/com/android/launcher3/SettingsActivity.java
@@ -21,6 +21,7 @@ import static com.android.launcher3.states.RotationHelper.getAllowRotationDefaul
 
 import android.annotation.TargetApi;
 import android.app.Activity;
+import android.app.ActivityManager;
 import android.app.AlertDialog;
 import android.app.Dialog;
 import android.app.DialogFragment;
@@ -110,7 +111,8 @@ public class SettingsActivity extends Activity {
                 getPreferenceScreen().removePreference(
                         findPreference(SessionCommitReceiver.ADD_ICON_PREFERENCE_KEY));
                 getPreferenceScreen().removePreference(iconBadgingPref);
-            } else if (!getResources().getBoolean(R.bool.notification_badging_enabled)) {
+            } else if (!getResources().getBoolean(R.bool.notification_badging_enabled)
+                    || getContext().getSystemService(ActivityManager.class).isLowRamDevice()) {
                 getPreferenceScreen().removePreference(iconBadgingPref);
             } else {
                 // Listen to system notification badge settings while this UI is active.
-- 
2.17.1

